tests/bdd/features/snapshot/list/test_list.py

```diff
@@ -0,0 +1,382 @@
+                    max_entries=1, starting_token=token
+                )
+                assert len(snapshots.entries) == 1
+                total_fetched += len(snapshots.entries)
+                if hasattr(snapshots, "next_token"):
+                    token = snapshots.next_token
+                else:
+                    break
+            assert total_fetched == len(VOLUME1_SNAP_IDS + VOLUME2_SNAP_IDS)
+
+        if filter == "by_volume":
+            # Fetch paginated - for a specfic volume
+            while token is not None:
+                snapshots = ApiClient.snapshots_api().get_volumes_snapshots(
+                    volume_id=VOLUME2_UUID, max_entries=1, starting_token=token
+                )
+                total_fetched += len(snapshots.entries)
+                if hasattr(snapshots, "next_token"):
+                    token = snapshots.next_token
+                else:
+                    break
+            assert total_fetched == len(VOLUME2_SNAP_IDS)
+    except openapi.exceptions.ApiException:
+        # Hitting this assert means test failed.
+        assert False, "Error in listing snapshots"
+
+
+def validate_snapshots(listed_snapshots, ref_snaps_list, state, volume):
+    """Validate the snapshots"""
+    assert len(listed_snapshots.entries) == len(ref_snaps_list)
+    for snapshot in listed_snapshots.entries:
+        try:
+            VOLUME1_SNAP_IDS.index(snapshot.state.uuid)
+            # assert snapshot.state.size == volume.spec.size
+            assert snapshot.state.clone_ready is False
+            timestamp_within(snapshot.state.timestamp)
+            wait_snapshot_state(snapshot, state)
+        except ValueError as ve:
+            assert False, "Invalid snapshot id found: " + snapshot.state.uuid
+
+
+def create_volume(replicas=1, volume_id="", publish_status="", replica_location=""):
+    if volume_id == "":
+        volume_id = VOLUME1_UUID
+
+    volume = ApiClient.volumes_api().put_volume(
+        volume_id,
+        CreateVolumeBody(
+            VolumePolicy(True),
+            replicas=replicas,
+            size=VOLUME_SIZE,
+            thin=False,
+        ),
+    )
+
+    if publish_status == "published":
+        node = NODE1
+        ApiClient.volumes_api().put_volume_target(
+            volume_id,
+            publish_volume_body=PublishVolumeBody({}, Protocol("nvmf"), node=node),
+        )
+
+    pytest.volume = volume
+    return volume
+
+
+def create_snapshots_for_volume(volume_id, snap_id_list):
+    for snapid in snap_id_list:
+        ApiClient.snapshots_api().put_volume_snapshot(volume_id, snapid)
+
+
+def restart_node(name):
+    Docker.restart_container(name)
+    wait_node_ready(name)
+
+
+@retry(wait_fixed=100, stop_max_attempt_number=20)
+def wait_node_ready(name):
+    node = ApiClient.nodes_api().get_node(name)
+    assert node.state.status == NodeStatus("Online")
+
+
+@retry(wait_fixed=500, stop_max_attempt_number=20)
```

Comment:
Keep the same as other bdds
```suggestion
@retry(wait_fixed=100, stop_max_attempt_number=20)
```